typed-racket-lib/typed-racket/utils/vector-contract.rkt

```diff
@@ -0,0 +1,12 @@
+#lang racket/base
+
+(require racket/contract)
+(provide immutable-vector/c immutable-vectorof/c mutable-vector/c mutable-vectorof/c)
+
+;; TODO if the element contracts are flat, we can pass the `#:flat` option
+;;  to `vector/c` and `vectorof` to make these combinators flat.
```

Comment:
would it not work to just do this TODO now -- i.e. throw in an `ormap` with `flat-contract?` and and use `#:flat? #t` if it's true?